src/super_gradients/training/datasets/samplers/class_balanced_sampler.py

```diff
@@ -0,0 +1,151 @@
+    :param oversample_threshold:        A frequency threshold (fraction, 0-1). Classes that are *less frequent* than this threshold will be oversampled.
+                                        The default value is None. If None, the median of the class frequencies will be used.
+    :param oversample_aggressiveness:   How aggressive the oversampling is. The higher the value, the more aggressive the oversampling is.
+                                        The default value is 0.5, and corresponds to the implementation in the paper.
+                                        A value of 0.0 corresponds to no oversampling.
+
+    Value of 1.0 corresponds to no oversampling.
+    """
+    if oversample_threshold is None:
+        oversample_threshold = np.median(class_frequencies)
+
+    result = np.ones_like(class_frequencies, dtype=np.float32)
+    for cls, cls_freq in enumerate(class_frequencies):
+        if cls_freq == 0 or cls_freq > oversample_threshold:
+            continue
+        result[cls] = (oversample_threshold / cls_freq) ** oversample_aggressiveness
+
+    return result
+
+
+class ClassBalancer:
+    @staticmethod
+    def get_sample_repeat_factors(
+        class_information_provider: HasClassesInformation,
+        oversample_threshold: Optional[float] = None,
+        oversample_aggressiveness: float = 0.5,
+    ) -> List[float]:
+        """
+        Oversampling scarce classes from detection dataset, following sampling strategy described in https://arxiv.org/pdf/1908.03195.pdf.
+
+        :param class_information_provider:      An object (probably a dataset) that provides the class information.
+        :param oversample_threshold:            A frequency threshold (fraction, 0-1). Classes that are *less frequent* than this threshold will be oversampled.
+                                                The default value is None. If None, the median of the class frequencies will be used.
+        :param oversample_aggressiveness:       How aggressive the oversampling is. The higher the value, the more aggressive the oversampling is.
+                                                The default value is 0.5, and corresponds to the implementation in the paper.
+                                                A value of 0.0 corresponds to no oversampling.
+
+
+        The repeat factor is computed as followed:
+        1. For each class c, compute the fraction # of images that contain it (its frequency): :math:`f(c)`
+        2. For each class c, compute the category-level repeat factor: :math:`r(c) = max(1, aggressiveness(threshold/f(c)))`
+        3. For each image I, compute the image-level repeat factor: :math:`r(I) = max_{c in I} r(c)`
+
+        Returns a list of repeat factors (length = dataset_length). How to read: result[i] is a float, indicates the repeat factor of image i.
+        """
+
+        class_information = class_information_provider.get_dataset_classes_information()
```

Comment:
A shape hint of `class_information` would help. E.g is it `[Num Samples, Num Classes] or vice versa

Reply:
Done, [db2f477](https://github.com/Deci-AI/super-gradients/pull/1865/commits/db2f477deb3b310c30c275e56eeaae7ef71979f2)